components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -100,30 +122,78 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
+            String allPaths = (String) constantObj;
+            modelStorageLocations = allPaths.split(",");
         } else {
-            throw new ExecutionPlanValidationException("ML model storage-location has not been defined as the first parameter");
+            throw new ExecutionPlanValidationException(
+                    "ML model storage-location has not been defined as the first parameter");
         }
 
         // data-type
         Attribute.Type outputDatatype;
-        if(attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+        if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
             Object constantObj = ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
             outputType = (String) constantObj;
             outputDatatype = getOutputAttributeType(outputType);
         } else {
-            throw new ExecutionPlanValidationException("Response variable type has not been defined as the second parameter");
+            throw new ExecutionPlanValidationException(
+                    "Response variable type has not been defined as the second parameter");
         }
 
-        try {
-            modelHandler = new ModelHandler(modelStorageLocation);
-            responseVariable = modelHandler.getResponseVariable();
-        } catch (Exception e) {
-            log.error("Error while retrieving ML-model : " + modelStorageLocation, e);
-            throw new ExecutionPlanCreationException("Error while retrieving ML-model : " + modelStorageLocation + "\n" + e.getMessage());
+        modelHandlers = new ModelHandler[modelStorageLocations.length];
+        for (int i = 0; i < modelStorageLocations.length; i++) {
+            try {
+                modelHandlers[i] = new ModelHandler(modelStorageLocations[i]);
+            } catch (ClassNotFoundException e) {
+                log.error("Error while retrieving ML-model : " + modelStorageLocations[i], e);
+                throw new ExecutionPlanCreationException(
+                        "Error while retrieving ML-model : " + modelStorageLocations[i] + "\n" + e.getMessage());
+            } catch (URISyntaxException e) {
+                log.error("Error while retrieving ML-model : " + modelStorageLocations[i], e);
+                throw new ExecutionPlanCreationException(
+                        "Error while retrieving ML-model : " + modelStorageLocations[i] + "\n" + e.getMessage());
+            } catch (MLInputAdapterException e) {
+                log.error("Error while retrieving ML-model : " + modelStorageLocations[i], e);
+                throw new ExecutionPlanCreationException(
+                        "Error while retrieving ML-model : " + modelStorageLocations[i] + "\n" + e.getMessage());
+            } catch (IOException e) {
+                log.error("Error while retrieving ML-model : " + modelStorageLocations[i], e);
+                throw new ExecutionPlanCreationException(
```

Comment:
Move the log and the exception throwing code to a helper method, s.t. you can avoid duplicating code.